Forms: Fix empty string submit validation

enejb opened the pull request:
Fixes FORM-NEW

Fixes issue where the value could be set to a space and the form would be considered non empty. 

## Proposed changes:
* Trim the value before comparing it on the JS side. 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
p1753989335777329-slack-C052XEUUBL4

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
1. Add the form. 
2. Fill out the form with space characters. 
3. Notice that you can't submit the form any more via a JS enabled browser.

enejb committed:
Forms: fix allowing spaces to be used as valid characters for non empty form

enejb committed:
changelog

CGastrell commented on projects/packages/forms/src/modules/form/view.js:
```diff
@@ -132,7 +132,7 @@ const { state } = store( NAMESPACE, {
 				return false;
 			}
-			return ! Object.values( context.fields ).some( field => field.value !== '' );
+			return ! Object.values( context.fields ).some( field => field.value.trim() !== '' );
```
Are we sure all values are strings there? should we `field.value?.trim?.()`?

enejb replied:
Yeah. Checkboxes are causing issues here. Fixing.

enejb committed:
Improved is Empty Checking

enejb committed:
Fix checkboxes is Required error even if they are not required

enejb committed:
Clear the showErrors when going back after submit

CGastrell commented on projects/packages/forms/src/modules/form/view.js:
```diff
@@ -13,6 +13,26 @@ import {
+	}
+
+	if ( Array.isArray( value ) && value.length === 0 ) {
+		return true;
```
I think we can use recursion here, like `return value.length === 0 || value.every( isEmptyValue )`

CGastrell commented on projects/packages/forms/src/modules/form/view.js:
```diff
@@ -13,6 +13,26 @@ import {
+	}
+
+	if ( typeof value === 'object' && Object.keys( value ).length === 0 ) {
+		return true;
```
Same as above

enejb committed:
Refactor empty value validation and improve tests

Moved and enhanced the isEmptyValue function for broader empty value detection, updated validateField to use it for required checks, and removed redundant code in form view. Added comprehensive unit tests for isEmptyValue and updated required field validation tests to cover more cases.

enejb committed:
Add more tests

CGastrell reviewed (approved):
Let's ship this, it's looking good on my end unless you want to add more tests, but those are passing fine right now
